src/bin/tikv-ctl.rs

```diff
@@ -355,21 +371,111 @@ fn dump_region_info(db: &DB, region_id: u64, skip_tombstone: bool) {
     println!("apply state: {:?}", apply_state);
 }
 
-fn dump_all_region_info(db: DB, skip_tombstone: bool) {
+fn convert_gbmb(mut bytes: u64) -> String {
+    let mb = if bytes % (1024 * 1024 * 1024) == 0 {
+        String::from("")
+    } else {
+        format!("{:.3} MB ",
+                (bytes % (1024 * 1024 * 1024)) as f64 / 1024.0 / 1024.0)
+    };
+    bytes /= 1024 * 1024 * 1024;
+    let gb = if bytes == 0 {
+        String::from("")
+    } else {
+        format!("{} GB ", bytes)
+    };
+    format!("{}{}", gb, mb)
+}
+
+fn dump_region_size(db: &DB, region_id: u64) {
+    println!("region id: {}", region_id);
+    let size = get_region_size(db, region_id);
+    println!("region size: {}", convert_gbmb(size));
+}
+
+fn dump_all_region_info(db: &DB, skip_tombstone: bool) {
+    let region_ids = get_all_region_id(db);
+    for region_id in region_ids {
+        dump_region_info(db, region_id, skip_tombstone);
+    }
+}
+
+fn dump_all_region_size(db: &DB) {
+    let region_ids = get_all_region_id(db);
+    let mut v: Vec<(u64, u64)> = Vec::new();
+    let mut total_size = 0;
+    let mut region_number = 0;
+    for region_id in region_ids {
+        let size = get_region_size(db, region_id);
+        v.push((region_id, size));
+        total_size += size;
+        region_number += 1;
```

Comment:
This code is more like C instead of Rust.
Try use iterator, Vec member functions.